addon/components/horizontal-bar-chart.js

```diff
@@ -230,13 +230,15 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     };
   }),
 
-  valueLabelAttrs: Ember.computed('xScale', 'barThickness', 'labelPadding', function() {
+  valueLabelAttrs: Ember.computed('xScale', 'barThickness', 'labelPadding', 'hasNegativeValues', 'hasPositiveValues', function() {
     var xScale = this.get('xScale');
     // Anchor the label 'labelPadding' away from the zero line
     // How to anchor the text depends on the direction of the bar
+
     return {
       x: (d) => {
-        if (d.value < 0) {
+        //Value is negative, or Value is 0 and all other data is negative
+        if ((d.value < 0) || ((d.value === 0) && this.get('hasNegativeValues') && !this.get('hasPositiveValues'))) {
```

Comment:
I didn't see it before, but functionally it will work well! I will swap it in

That being said, technically it is a misnomer is 0 isn't "negative" or "positive". Therefore, "hasAllNegativeValues" really should be "hasNoPositiveValues" if all data <= 0. 

"hasAllNegativeValues" implies that no data point is 0. We should probably change that name later